### PR TITLE
Change to named parameters and add "req", needed for Google OAuth

### PR DESCRIPTION
**Pull request by bmeck**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,21 +8,32 @@ Passport strategy for authenticating an `npm` client.
 const { NPMStrategy, NPMStrategyErrorHandler } = require('passport-npm');
 // check the login of a user and create a user object
 // set user to `false` if login is invalid
-function authenticate(username, password, done) {
+function authenticate({
```
Comment (jcrugzz) [inline README.md:11]: should we unpack these from `options` to support < node@6 or does it matter?

File: README.md [hunk 2]
```diff
@@ -8,21 +8,32 @@ Passport strategy for authenticating an `npm` client.
 const { NPMStrategy, NPMStrategyErrorHandler } = require('passport-npm');
```
Comment (jcrugzz) [inline README.md:8]: i guess this applies here as well

File: README.md [hunk 3]
```diff
@@ -8,21 +8,32 @@ Passport strategy for authenticating an `npm` client.
...
+}, done) {
   done(null, /* User on success || false on failure */);
 }
 // creates a string for `npm` to store in user .npmrc
 // send a falsey string to not use tokens (not recommended)
 //
 // use this to prevent `npm` from storing username and password on disk
 // commonly used to store an access token
-function serializeNPMToken(username, password, done) {
+function serializeNPMToken({
```
Comment (jcrugzz) [inline README.md:23]: and here

File: README.md [hunk 4]
```diff
@@ -8,21 +8,32 @@ Passport strategy for authenticating an `npm` client.
...
+  name, // string
+  password // string
+}, done) {
   done(null, /* non-falsey token string || false to store username and password instead */)
 }
 // similar to `authenticate`
 //
 // consumes the result token string serializeToken
-function deserializeNPMToken(tokenString, done) {
+function deserializeNPMToken({
```
Comment (jcrugzz) [inline README.md:33]: and here

File: README.md (hunk 1 above)
Comment (bmeck) [reply README.md:11]: makes sense, though we do it all over, v4 support is on for quite a while

**Comment by jcrugzz** [comment]
@bmeck seems fine to me. Since we are using named parameters we do limit ourselves to node@6 so we should make a decision whether we should support node@4. Otherwise 👍

**Pull request merged**